utils.go

```diff
@@ -74,7 +74,7 @@ func GetRootlessRuntimeDir(rootlessUid int) (string, error) {
 	if runtimeDir == "" {
 		tmpDir := fmt.Sprintf("/run/user/%d", rootlessUid)
 		st, err := system.Stat(tmpDir)
-		if err == nil && int(st.UID()) == os.Getuid() && st.Mode() == 0700 {
+		if err == nil && int(st.UID()) == os.Getuid() && st.Mode()&0700 == 0700 {
```

Comment:
we want to check that the directory is accessible only by the current user.  Now the `&` mask drops the check on the other users. 
We could relax it to accept `0711`, that is the mode we use when using user namespaces, but I don't think we should allow a directory writeable by other users.

Reply:
The problem is the st.Mode() != 0700 So this always fails.  I was thinking this would fix this, but it fails.

Just unset XDG_RUNTIME_DIR, and even if /run/usr/UID is setup correctly 0700, Golang reports this as a different numer.  So we fail over to /tmp/UID.